terminal.cpp

```diff
@@ -174,19 +174,21 @@ void Terminal::putString(QString str, bool unEscape)
         iPtyIFace->writeTerm(str);
 }
 
-void Terminal::keyPress(int key, int modifiers)
+void Terminal::keyPress(int key, int modifiers, QString text)
 {
     QChar c(key);
 
     resetBackBufferScrollPos();
 
+    if ((!text.isEmpty()) && (text[0].unicode() == 0)) text.clear();
```

Comment:
Looks strange case, not sure if Qt is really supposed to pass such string. KeyEvent documentation says text is empty with modifier keys. Anyway, maybe text[0].isNull() could be slightly clearer?

@giucam any chance QtWayland passes bad event data?